### DISCUSSION TITLE
Commonsense Knowledge: Project Ideas

### DISCUSSION THREAD

**Discussion by mayanks43** [eduardoblanco/csc696H-Spring24#15] [Ideas]
Please submit your project ideas here.

**Comment by devsawgrad** [comment]
(Not sure if these are even project ideas, would like to know thoughts on these)
- The critic model in the Symbolic knowledge distillation paper seems to be crucial for their performance as observed. Instead of training the critic model on the machine generated data, training it on the **original human annotated corpus** and then evaluating the datasets/model. Would that be a better gauge on how well the critic model works? and produce potentially better data sets without performing extra manual validation which was done in the paper?
-  Do the same method in the paper for datasets in other tasks like Inference, Question Answering etc to see if it works well.
- Try out different Critic models on different  annotated (or preferably machine generated from a sample) datasets and investigate the effect on diversity.

**Comment by mayanks43** [comment]
Using LLMs for self-critique
=====================
[This paper](https://aclanthology.org/2023.emnlp-main.81/) expands the critic model idea from the Symbolic Knowledge Distillation paper into its own paper. Something interesting to understand would be if LLMs can self critique themselves and is that enough to do away with this critic model for commonsense knowledge verification. I was able to get GPT-4 to self critique itself to correct its answer to the question "does marble float in a bowl of mercury?" by simply giving the prompt: "Critic your previous output". Does this process generalize to other commonsense verification tasks it currently fails on? That would be a good question to investigate.
